### PR TITLE
fix clean stop behavior when consumer queue is waiting on new events from kafka

### PR DESCRIPTION
**Pull request by talevy**

### CODE REVIEW

File: spec/inputs/kafka_spec.rb [hunk 1]
```diff
@@ -42,7 +42,7 @@ def run(a_num_threads, a_queue)
     Thread.new do
       while true
         a_queue << blah
-        sleep 0.2
+        sleep 10
```
Comment (talevy) [inline spec/inputs/kafka_spec.rb]: with this increased sleep time, test fails without the fix in `kafka.rb`

File: lib/logstash/inputs/kafka.rb [hunk 2]
```diff
@@ -155,6 +158,7 @@ def run(logstash_queue)
 
   public
   def stop
+    @kafka_client_queue.push(LogStash::SHUTDOWN)
```
Comment (andrewvc) [inline lib/logstash/inputs/kafka.rb]: I'm not so sure we should repurpose the LogStash::SHUTDOWN object here. I could go either way on this, but this is more of a kafka shutdown. Maybe it should be some other constant? Anyway, if you feel like using the code as is go for it, but do think about it.

Comment (talevy) [inline lib/logstash/inputs/kafka.rb]: hmmm. sounds fair to keep `SHUTDOWN` only to the main logstash queue. I'll go ahead and introduce a new constant

**Comment by andrewvc** [comment]
LGTM, but please read my comment above before merging.

**Comment by talevy** [comment]
updated, let me know what you think

**Pull request merged**